hypervisor/arch/x86/rdt.c

```diff
@@ -80,14 +80,9 @@ static void init_cat_capability(int res)
 #else
 	res_cap_info[res].res.cache.is_cdp_enabled = false;
 #endif
+	res_cap_info[res].num_closids = (uint16_t)(edx & 0xffffU) + 1U;
 	if (res_cap_info[res].res.cache.is_cdp_enabled) {
-		res_cap_info[res].num_closids = (uint16_t)((edx & 0xffffU) >> 1U) + 1U;
-		/* enable CDP before setting COS to simplify CAT mask remapping
-		 * and prevent unintended behavior.
-		 */
-		msr_write(res_cap_info[res].res.cache.msr_qos_cfg, 0x1UL);
-	} else {
-		res_cap_info[res].num_closids = (uint16_t)(edx & 0xffffU) + 1U;
+		res_cap_info[res].num_closids /= 2;
```

Comment:
“/” may be complicated.  Can you use ">>= 1" ?

Reply:
yes, will change

Reply:
fixed and updated